src/FsCheck/Gen.fs

```diff
@@ -313,6 +313,20 @@ module Gen =
     [<CompiledName("ListOf")>]
     let listOfLength n arb = sequence [ for _ in 1..n -> arb ]
 
+    ///Generates a random permutation of the given sequence.
+    //[category: Creating generators]
+    [<CompiledName("Shuffle")>]
+    let shuffle xs =
+        gen {
+            let h = 922337203
```

Comment:
Not being able to use int.MinValue and int.MaxValue is indeed a bug in the random implementation. As you know I intend to replace it for v3.0...